SmurfEngine/Items/Inventory.cs

```diff
@@ -8,12 +9,40 @@ public class Inventory
 
         public void Add(Item item, int quantity)
         {
+            if (quantity == 0) return;
+
             if (this.Contents.TryGetValue(item.Name, out var invItem))
                 invItem.Quantity += quantity;
             else
                 this.Contents.Add(item.Name, new InventoryItem(item, quantity));
         }
 
+        public bool Remove(Item item, int quantity)
+        {
+            if (quantity <= 0) return false;
+
+            if (this.Contents.TryGetValue(item.Name, out var invItem))
+            {
+                if (quantity >= invItem.Quantity)
+                    this.Contents.Remove(item.Name);
+                else
+                    invItem.Quantity -= quantity;
+
+                return true;
+            }
+            
+            return false;
+        }
+
+        public bool Check(Item item, int quantity = 1)
+        {
+            if (this.Contents.TryGetValue(item.Name, out var invItem))
+            {
```

Comment:
I think the curly brace here should be removed.  We have single line if statements elsewhere and we don't have curly braces.